packages/example-chat-system/browser-client/service-chat.ts

```diff
@@ -24,25 +24,26 @@ export interface ChatEvent {
      * @generated from protobuf oneof: event
      */
     event: {
-        oneofKind: "joined";
+        kind: "joined";
         /**
          * @generated from protobuf field: string joined = 2;
          */
-        joined: string;
+        value: string;
     } | {
-        oneofKind: "message";
+        kind: "message";
         /**
          * @generated from protobuf field: string message = 3;
          */
-        message: string;
+        value: string;
     } | {
-        oneofKind: "left";
+        kind: "left";
         /**
          * @generated from protobuf field: string left = 4;
          */
-        left: string;
+        value: string;
     } | {
-        oneofKind: undefined;
+        kind: undefined;
+        value?: never;
```

Comment:
Do we need `never` here (and in `PartialMessage`)? I imagine `undefined` would be a bit more forgiving with user input, and still do the same job - unless I'm missing something.

Reply:
There is a difference if you have `[exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes)` enabled.

```ts
declare const foo: { value?: never };
declare const bar: { value?: undefined };

foo.value = undefined; // error
bar.value = undefined; // no error
```

The idea being that if you receive the `{ value?: never }` type you can read the `value` property and it can be `undefined` (or rather, unset), but you can never assign a value to it. I think this is the more correct type for an unknown oneof group. At the very least I we need to keep the `value` property in the object type when `kind` is undefined so that the new `PartialMessage` case still matches correctly.

>I imagine `undefined` would be a bit more forgiving with user input

I suppose that is true, but I'm also not sure what the use case is for that. We still have the helper functions in `oneof.ts` that can be used if direct mutation is required.